pg_search/src/index/fast_fields_helper.rs

```diff
@@ -46,19 +46,22 @@ impl FFHelper {
                 let mut lookup = Vec::new();
                 for field in fields {
                     match field {
-                        WhichFastField::Ctid
-                        | WhichFastField::TableOid
-                        | WhichFastField::Score
-                        | WhichFastField::Junk(_) => lookup.push((
-                            fast_fields_reader.clone(),
-                            String::from("junk"),
-                            OnceLock::from(FFType::Junk),
-                        )),
-                        WhichFastField::Named(name, _) => lookup.push((
+                        Some(WhichFastField::Named(name, _)) => lookup.push((
                             fast_fields_reader.clone(),
                             name.to_string(),
                             OnceLock::default(),
                         )),
+                        Some(
+                            WhichFastField::Ctid
+                            | WhichFastField::TableOid
+                            | WhichFastField::Score
+                            | WhichFastField::Junk(_),
+                        )
+                        | None => lookup.push((
+                            fast_fields_reader.clone(),
+                            String::from("junk"),
+                            OnceLock::from(FFType::Junk),
+                        )),
```

Comment:
Could you add a comment explaining the implications of the `None` case? It looks like it will happen when `align_fast_fields_with_tuple_descriptor` fails to find a field at execution time which we expected to be there at planning time.

Do we know when that occurs? And can we be sure that it won't cause a crash/failure to find a field? From inspection, it seems like it might cause us to fail to populate a fast field in `extract_data_from_fast_fields`, which would result in ... `null`, I think?

Reply:
The assumption is that if any element of `fields` is `None`, it's going to be treated as Junk. It only happens for the fields that are not marked as FastField via https://github.com/paradedb/paradedb/blob/aceeebbcf92011692d1ff86afa98696240c2fd16/pg_search/src/postgres/customscan/pdbscan/exec_methods/fast_fields/mod.rs#L96-L97

Reply:
Will add a comment.

Reply:
Done in [8c3e5fc](https://github.com/paradedb/paradedb/pull/2554/commits/8c3e5fc5628cc3b82f3b3ee555c4ecb82158b040)

Reply:
Hm... yea, a `NULL` doesn't seem like what we want. But if this behind a feature flag, then ok for now.